File: src/main/java/me/andre111/d20server/scripting/variable/parser/VariableParserParentTokenSelected.java
package me.andre111.d20server.scripting.variable.parser;

import me.andre111.d20server.scripting.ParserContext;
import me.andre111.d20server.scripting.TokenFinder;
import me.andre111.d20server.scripting.TokenFinderSelected;

public class VariableParserParentTokenSelected extends VariableParserParent {
	private static final TokenFinder FINDER = new TokenFinderSelected();
	
	@Override
	protected void updateContext(ParserContext context, String name) {
		context.setTokenFinder(FINDER);
	}
}

File: core/common/scripting/variable/parser/variable-parser-parent-token-selected.js
import { VariableParserParent } from './variable-parser-parent.js';
import { EntityFinderSelectedToken } from '../../finder/entity-finder-selected-token.js';

export class VariableParserParentTokenSelected extends VariableParserParent {
    static finder = new EntityFinderSelectedToken();

    constructor() {
        super();
    }

    updateContext(context, name) {
        context.setEntityFinder('token', VariableParserParentTokenSelected.finder);
    }
}